egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -139,7 +149,48 @@ def AddConvolutionLayer(config_lines, name, input,
     num_y_steps = (1 + (input_y_dim - filt_y_dim) / filt_y_step)
     output_dim = num_x_steps * num_y_steps * num_filters;
     return {'descriptor':  '{0}_conv_t'.format(name),
-            'dimension': output_dim}
+            'dimension': output_dim,
+            '3d-dim': [num_x_steps, num_y_steps, num_filters],
+            'vectorization': 'zyx'}
+
+
+def AddMaxpoolingLayer(config_lines, name, input,
+                      input_x_dim, input_y_dim, input_z_dim,
+                      pool_x_size, pool_y_size, pool_z_size,
+                      pool_x_step, pool_y_step, pool_z_step):
+    assert(input['dimension'] == input_x_dim * input_y_dim * input_z_dim)
```

Comment:
Better add a comment here stating that the Maxpool component assumes that the input is vectorized in the zyx format.